web-app/src/main/webapp/script.js

```diff
@@ -59,14 +62,28 @@ function showMessageOnInfoWindow(message, position, map, infoWindow) {
 window.onload = () => {
   // document.getElementById("report-form").style.display = "none";
   document.getElementById('report-button').addEventListener('click', showReportForm);
-  document.getElementById('submit-button').addEventListener('click', postUserReport);
+  document.getElementById('submit-button').addEventListener('click', geocodeAddress);
 };
 
 function showReportForm() {
   document.getElementById("report-form").style.display = "block";
 }
 
-function reportFormToURLQuery() {
+// This currently gets the address from the report form's location field (no autopopulate, no map picker)
+function geocodeAddress() {
```

Comment:
nit I think the aim of this function should be just convert address to latitude/longitude without doing extra stuff like generating report or posting it. If the status is OK, you can return a valid long/lat location object, otherwise, you can return values that is out of the range.